### PR TITLE
Allow filtering namespaces to create the ConfigMap istio-ca-root-cert

### PR DESCRIPTION
**Pull request by alandiegosantos**
This MR adds a parameter to filter namespaces in which the ConfigMap istio-ca-root-cert is created.
This parameter is validate on Complete() in [options.go](cmd/app/options/options.go) and propagated to
the [controller](pkg/controller/configmap.go).

So, the user can configure istio-csr to create the istio-root-ca-cert ConfigMap only on namespaces labelled _istio-csr-injection=enabled_ by adding the command-line parameter _--namespace-selector=istio-csr-injection=enabled_

### CODE REVIEW

**Comment by alandiegosantos** [comment]
Closes #125

File: cmd/app/options/options.go [hunk 1]
```diff
@@ -221,4 +225,9 @@ func (o *Options) addControllerFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.Controller.LeaderElectionNamespace,
 		"leader-election-namespace", "istio-system",
 		"Namespace to use for controller leader election.")
+
+	fs.StringVar(&o.Controller.NamespaceSelector,
+		"namespace-selector", "",
```
Comment (JoshVanL) [inline cmd/app/options/options.go:230]: Small nit: can we change `namespace-selector` to `configmap-namespace-selector` to make it more explicit what we are doing with these namespaces at glance?

**Review by JoshVanL — COMMENTED** [review]
Thanks @alandiegosantos!

All looks good to me, just a nit on the flag name.

/approve
/ok-to-test

File: cmd/app/options/options.go (hunk 1 above)
Comment (alandiegosantos) [reply cmd/app/options/options.go]: @JoshVanL The parameter is renamed. Thanks for the suggestion.

**Comment by JoshVanL** [comment]
Cheers!
/lgtm

**Pull request merged**